File: backend/src/main/java/com/equisoft/dca/backend/project/model/SystemType.java
package com.equisoft.dca.backend.project.model;

import com.equisoft.dca.backend.common.model.BaseEnum;

import lombok.Getter;

@Getter
public enum SystemType implements BaseEnum<Integer> {

	SOURCE(1, "Source"),
	TARGET(2, "Target"),
	UTILITY_SYSTEM(3, "Utility system");

	private final int id;

	private final String name;

	SystemType(int id, String name) {
		this.id = id;
		this.name = name;
	}

	@Override
	public Integer getId() {
		return id;
	}
}

File: react/src/lib/domain/entities/SystemType.ts
import BaseEnum from './BaseEnum';

export default class SystemType extends BaseEnum {
    public static SOURCE = new SystemType('1', 'system:systemTypeEnum:source');
    public static TARGET = new SystemType('2', 'system:systemTypeEnum:target');
    public static UTILITY_SYSTEM = new SystemType('3', 'system:systemTypeEnum:utility');

    public constructor(id?: string, name?: string) {
        super(id, name);
    }
}
